packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/orchestrate/send-process-retry.ts

```diff
@@ -0,0 +1,120 @@
+import { isRetryableError } from "../process/error";
+import { SamlCertsAndKeys } from "../../../saml/security/types";
+import {
+  OutboundDocumentRetrievalResp,
+  OutboundDocumentQueryResp,
+} from "@metriport/ihe-gateway-sdk";
+import { SignedDqRequest } from "../create/iti38-envelope";
+import { SignedDrRequest } from "../create/iti39-envelope";
+import { sendSignedDqRequest } from "../send/dq-requests";
+import { sendSignedDrRequest } from "../send/dr-requests";
+import { processDqResponse } from "../process/dq-response";
+import { processDrResponse } from "../process/dr-response";
+import { out } from "../../../../../../util/log";
+
+const { log } = out("IHE Gateway V2");
+
+function calculateBackoff(attempt: number, baseDelay = 2000, jitterRange = 2000): number {
+  const baseBackoffTime = Math.pow(2, attempt + 1) * baseDelay;
+  const jitter = (Math.random() - 0.5) * jitterRange;
+  return baseBackoffTime + jitter;
+}
+
+export async function sendProcessRetryRequests<T, R>({
+  signedRequest,
+  samlCertsAndKeys,
+  patientId,
+  cxId,
+  index,
+  sendRequest,
+  processResponse,
+  maxRetries = 2,
+}: {
+  signedRequest: T;
+  samlCertsAndKeys: SamlCertsAndKeys;
+  patientId: string;
+  cxId: string;
+  index: number;
+  sendRequest: (params: {
+    request: T;
+    samlCertsAndKeys: SamlCertsAndKeys;
+    patientId: string;
+    cxId: string;
+    index: number;
+  }) => Promise<R>;
+  processResponse: (params: {
+    response: R;
+    attempt: number;
+  }) => Promise<OutboundDocumentQueryResp | OutboundDocumentRetrievalResp>;
+  maxRetries?: number;
+}): Promise<OutboundDocumentQueryResp | OutboundDocumentRetrievalResp> {
+  let attempt = 0;
+
+  while (attempt < maxRetries) {
+    const response = await sendRequest({
+      request: signedRequest,
+      samlCertsAndKeys,
+      patientId,
+      cxId,
+      index,
+    });
+    const result = await processResponse({
+      response,
+      attempt,
+    });
+
+    if (!isRetryableError(result)) {
+      return result;
+    }
```

Comment:
This is confusing: we execute a function, get the response, and check if the error is "retryable" or not; if not, we return, if it is, we try again. What about the success case? Is it part of not retryable error?

Reply:
I can rename it isRetryable instead. 

In the success case the isRetryableError function checks if it has an error and then it doesnt so it doesnt retry